From 21e65dd7e309a476bcfa65c771ad07f768a03dc1 Mon Sep 17 00:00:00 2001
From: Patrick Yavitz <pyavitz@xxxxx.com>
Date: Thu, 1 Aug 2024 06:04:26 -0400
Subject: [PATCH] rtw88: decrease the log level of tx report

Reduce 'failed to get tx report from firmware' dmesg spam.
https://lore.kernel.org/linux-wireless/20210713104524.47101-1-pkshih@realtek.com/
https://cgit.freebsd.org/src/commit/?id=e140d551b78670fbf99c83a59438cb13de50420f

Signed-off-by: Patrick Yavitz <pyavitz@xxxxx.com>
---
 drivers/net/wireless/realtek/rtw88/tx.c | 12 +++++++-----
 1 file changed, 7 insertions(+), 5 deletions(-)

diff --git a/drivers/net/wireless/realtek/rtw88/tx.c b/drivers/net/wireless/realtek/rtw88/tx.c
index 1d8af54a2739..39c0acd786d8 100644
--- a/drivers/net/wireless/realtek/rtw88/tx.c
+++ b/drivers/net/wireless/realtek/rtw88/tx.c
@@ -205,14 +205,16 @@ void rtw_tx_report_purge_timer(void *cntx)
 	struct rtw_tx_report *tx_report = &rtwdev->tx_report;
 	unsigned long flags;
 
-	if (skb_queue_len(&tx_report->queue) == 0)
-		return;
-
-	rtw_warn(rtwdev, "failed to get tx report from firmware\n");
+	uint32_t qlen;
 
 	spin_lock_irqsave(&tx_report->q_lock, flags);
-	skb_queue_purge(&tx_report->queue);
+	qlen = skb_queue_len(&tx_report->queue);
+	if (qlen > 0)
+		skb_queue_purge(&tx_report->queue);
 	spin_unlock_irqrestore(&tx_report->q_lock, flags);
+
+	rtw_dbg(rtwdev, RTW_DBG_TX, "failed to get tx report from firmware: "
+		"txreport qlen %u\n", qlen);
 }
 
 void rtw_tx_report_enqueue(struct rtw_dev *rtwdev, struct sk_buff *skb, u8 sn)
-- 
2.39.2

